src/lava_callback.py

```diff
@@ -187,9 +202,194 @@ async def callback(node_id: str, request: Request):
     return 'OK', 202
 
 
+def decode_jwt(jwtstr):
+    '''
+    JWT secret stored at SETTINGS['jwt']['secret']
+    which means secret.toml file should have jwt section
+    with parameter secret= "<secret>"
+    '''
+    secret = SETTINGS.get('jwt', {}).get('secret')
+    if not secret:
+        logging.error('No JWT secret configured')
+        return None
+    return jwt.decode(jwtstr, secret, algorithms=['HS256'])
+
+
+def validate_permissions(jwtoken, permission):
+    if not jwtoken:
+        return False
+    try:
+        decoded = decode_jwt(jwtoken)
+    except Exception as e:
+        logging.error(f'Error decoding JWT: {e}')
+        return False
+    if not decoded:
+        logging.error('Invalid JWT')
+        return False
+    permissions = decoded.get('permissions')
+    if not permissions:
+        logging.error('No permissions in JWT')
+        return False
+    if permission not in permissions:
+        logging.error(f'Permission {permission} not in JWT')
+        return False
+    return decoded
+
+
+def find_parent_kind(node, api_helper, kind):
+    parent_id = node.get('parent')
+    if not parent_id:
+        return None
+    parent_node = api_helper.api.node.get(parent_id)
+    if not parent_node:
+        return None
+    if parent_node.get('kind') == kind:
+        return parent_node
+    return find_parent_kind(parent_node, api_helper, kind)
+
+
+@app.post('/api/jobretry')
```

Comment:
This is a second set of APIs implemented in a pipeline service, bypassing the main API.

Does it do its job? It most likely works
Would I support such a shift in a solution architecture? It most likely works